### PR TITLE
PACT-1005 Modify the request & reply model for Agents to match the API services model

### PR DESCRIPTION
**Pull request by userjd**

### CODE REVIEW

File: app/uk/gov/nationalarchives/omega/editorial/models/AgentDescription.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import play.api.libs.json.{ Format, __ }
+
+case class AgentDescription(
+  identifier: String,
+  label: String,
+  authorityFile: Option[Boolean],
+  depository: Option[Boolean],
+  versionTimestamp: String,
+  dateFrom: Option[String],
+  dateTo: Option[String],
```
Comment (rwalpole) [inline app/uk/gov/nationalarchives/omega/editorial/models/AgentDescription.scala:34]: We should be consistent and use a default of `None` for `dateFrom` and `dateTo` as well.

File: app/uk/gov/nationalarchives/omega/editorial/models/GetAgentSummaryList.scala [hunk 2]
```diff
@@ -21,11 +21,22 @@
...
 
-import play.api.libs.json.{ Format, Json }
+import play.api.libs.functional.syntax.{ toFunctionalBuilderOps, unlift }
+import play.api.libs.json.{ Format, __ }
 
-case class GetAgentSummaryList(agentTypeList: List[AgentType], depository: Option[Boolean] = None)
+case class GetAgentSummaryList(
+  agentType: List[AgentType],
+  versionTimestamp: Option[String] = None,
+  depository: Option[Boolean] = Some(false),
```
Comment (rwalpole) [inline app/uk/gov/nationalarchives/omega/editorial/models/GetAgentSummaryList.scala:30]: The default should be `None`.

File: app/uk/gov/nationalarchives/omega/editorial/models/GetAgentSummaryList.scala [hunk 3]
```diff
@@ -21,11 +21,22 @@
...
+  authorityFile: Option[Boolean] = Some(false)
```
Comment (rwalpole) [inline app/uk/gov/nationalarchives/omega/editorial/models/GetAgentSummaryList.scala:31]: The default should be `None`.

File: app/uk/gov/nationalarchives/omega/editorial/models/AgentDescription.scala (hunk 1 above)
Comment (userjd) [reply app/uk/gov/nationalarchives/omega/editorial/models/AgentDescription.scala]: Done

File: app/uk/gov/nationalarchives/omega/editorial/models/GetAgentSummaryList.scala (hunk 2 above)
Comment (userjd) [reply app/uk/gov/nationalarchives/omega/editorial/models/GetAgentSummaryList.scala]: Done

File: app/uk/gov/nationalarchives/omega/editorial/models/GetAgentSummaryList.scala (hunk 3 above)
Comment (userjd) [reply app/uk/gov/nationalarchives/omega/editorial/models/GetAgentSummaryList.scala]: Done

**Pull request merged**